src/main/java/de/allcom/services/WishlistService.java

```diff
@@ -0,0 +1,82 @@
+package de.allcom.services;
+
+import de.allcom.dto.product.ProductWishlistDto;
+import de.allcom.exceptions.RestException;
+import de.allcom.models.Product;
+import de.allcom.models.ProductImage;
+import de.allcom.models.Wishlist;
+import de.allcom.models.WishlistItem;
+import de.allcom.repositories.ProductRepository;
+import de.allcom.repositories.WishlistItemRepository;
+import de.allcom.repositories.WishlistRepository;
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class WishlistService {
+    private final WishlistRepository wishlistRepository;
+    private final WishlistItemRepository wishlistItemRepository;
+    private final ProductRepository productRepository;
+    private final WishlistItemService wishlistItemService;
+
+    public List<ProductWishlistDto> findProducts(Long userId) {
```

Comment:
What if the user has 100 products in their wishlist? I think Page<ProductWishlistDto> is needed here